tools.sh

```diff
@@ -42,6 +43,22 @@ if test -z ${YQ-}; then
     YQ=./yq
 fi
 
+# Download butane for creating ignition config
+if test -z ${BUTANE-}; then
+    echo "Downloading butane binary to create ignition config"
+    curl -L https://github.com/coreos/butane/releases/download/v0.14.0/butane-${ARCH}-unknown-linux-gnu -o butane
```

Comment:
Did you check with the coreos people if they could build aarch64 container images instead?

Reply:
@cfergeau yes I did check with them and response I got `afaik coreos/ ones are auto-built directly by Quay. So the set of architectures depends on which auto-builders exist.` An on quay the auto-builders are x86_64 one.

Reply:
We need to check the GPG signature of these binaries to have some validation regarding the origin of these binaries.

Reply:
Alternatively:
```
$ yum info butane
Available Packages
Name         : butane
Version      : 0.13.1
Release      : 1.fc35
Architecture : x86_64
Size         : 2.0 M
Source       : butane-0.13.1-1.fc35.src.rpm
Repository   : fedora
Summary      : Butane config transpiler
URL          : https://github.com/coreos/butane
License      : ASL 2.0
Description  : 
             : Butane translates human-readable Butane Configs into
             : machine-readable Ignition configs for provisioning operating
             : systems that use Ignition.

$ dnf info ignition-validate
Available Packages
Name         : ignition-validate
Version      : 2.13.0
Release      : 3.fc35
Architecture : x86_64
Size         : 1.4 M
Source       : ignition-2.13.0-3.fc35.src.rpm
Repository   : updates
Summary      : Validation tool for Ignition configs
URL          : https://github.com/coreos/ignition
License      : ASL 2.0
Description  : Ignition is a utility used to manipulate systems during the
             : initramfs. This includes partitioning disks, formatting
             : partitions, writing files (regular files, systemd units, etc.),
             : and configuring users. On first boot, Ignition reads its
             : configuration from a source of truth (remote URL, network
             : metadata service, hypervisor bridge, etc.) and applies the
             : configuration.
             : 
             : This package contains a tool for validating Ignition
             : configurations.
```

This is consistent with ` sudo yum -y install /usr/bin/coreos-installer`

Reply:
@cfergeau This is not available on the RHEL and since our CI uses RHEL system it is going to fail.

Reply:
I can see `butane-0.14.0-1.rhaos4.10.el8` and `ignition-2.13.0-3.rhaos4.11.el8` in RHEL, maybe the CI system could be subscribed to these additional repos?

Reply:
I followed this https://docs.openshift.com/container-platform/4.10/machine_management/more-rhel-compute.html to get access to ignition-validate and butane:
```
$ sudo subscription-manager attach --pool=xxxx
$ sudo subscription-manager repos  --enable="rhocp-4.10-for-rhel-8-x86_64-rpms"
$ yum info ignition-validate butane
Not root, Subscription Management repositories not updated
Last metadata expiration check: 0:02:07 ago on Fri May 13 13:47:43 2022.
Available Packages
Name         : butane
Version      : 0.14.0
Release      : 1.rhaos4.10.el8
Architecture : x86_64
Size         : 2.0 M
Source       : butane-0.14.0-1.rhaos4.10.el8.src.rpm
Repository   : rhocp-4.10-for-rhel-8-x86_64-rpms
Summary      : Butane config transpiler
URL          : https://github.com/coreos/butane
License      : ASL 2.0
Description  : 
             : Butane translates human-readable Butane Configs into machine-readable Ignition
             : configs for provisioning operating systems that use Ignition.

Name         : ignition-validate
Version      : 2.13.0
Release      : 2.rhaos4.10.el8
Architecture : x86_64
Size         : 1.4 M
Source       : ignition-2.13.0-2.rhaos4.10.el8.src.rpm
Repository   : rhocp-4.10-for-rhel-8-x86_64-rpms
Summary      : Validation tool for Ignition configs
URL          : https://github.com/coreos/ignition
License      : ASL 2.0
Description  : Ignition is a utility used to manipulate systems during the initramfs.
             : This includes partitioning disks, formatting partitions, writing files
             : (regular files, systemd units, etc.), and configuring users. On first
             : boot, Ignition reads its configuration from a source of truth (remote
             : URL, network metadata service, hypervisor bridge, etc.) and applies
             : the configuration.
             : 
             : This package contains a tool for validating Ignition configurations.
```